### ISSUE TITLE
TASK Purge PHP version packages. Fails

### ISSUE THREAD

**Issue by mglaman** [geerlingguy/drupal-vm#1319]
### Issue Type

 - Bug Report

### Your Environment

Local

```
$ vboxmanage --version
5.1.18r114002

$ ansible --version
ansible 2.3.0.0
  config file = /Users/mglaman/Clients/commerceguys/reports-leancom/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

Production

```

```

### Your OS

 - **macOS** 10.12.4
 - **Linux** ubuntu 16

### Full console output

```
DRUPALVM_ENV=prod ansible-playbook -i vm/inventory vendor/geerlingguy/drupal-vm/provisioning/playbook.yml -e "config_dir=$(pwd)/vm" --sudo --tags=drupal --extra-vars "ansible_become_pass=${ANSIBLE_VAULT_PASSWORD}"

PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************
ok: [104.236.244.212]

TASK [Define config_dir.] ******************************************************
skipping: [104.236.244.212]

TASK [Include optional configuration files.] ***********************************
ok: [104.236.244.212] => (item=/home/ubuntu/dashboard/vm/config.yml)
ok: [104.236.244.212] => (item=/home/ubuntu/dashboard/vm/prod.config.yml)

TASK [Define fully qualified domain name.] *************************************
ok: [104.236.244.212]

TASK [Define short hostname.] **************************************************
ok: [104.236.244.212]

TASK [Add hostname to /etc/hosts.] *********************************************
ok: [104.236.244.212]

TASK [Configure hostname.] *****************************************************
ok: [104.236.244.212]

TASK [Set the hostname for current session.] ***********************************
skipping: [104.236.244.212]

TASK [include] *****************************************************************
included: /home/ubuntu/dashboard/vendor/geerlingguy/drupal-vm/provisioning/tasks/init-debian.yml for 104.236.244.212

TASK [Update apt cache if needed.] *********************************************
ok: [104.236.244.212]

TASK [Install required dependencies.] ******************************************
ok: [104.236.244.212] => (item=[u'curl', u'python-apt', u'python-pycurl', u'sudo', u'unzip', u'make'])

TASK [Configure /etc/mailname.] ************************************************
ok: [104.236.244.212]

TASK [Define php_xhprof_html_dir.] *********************************************
skipping: [104.236.244.212]

TASK [Add repository for Apache 2.4.9+ (Ubuntu 12/14).] ************************
skipping: [104.236.244.212]

TASK [Add repository for PHP versions (Ubuntu).] *******************************
ok: [104.236.244.212]

TASK [Add repository for PHP 5 compatibility packages (Ubuntu).] ***************
skipping: [104.236.244.212]

TASK [Set the correct opcache filename (Ubuntu).] ******************************
ok: [104.236.244.212]

TASK [Add dependencies for PHP versions (Debian).] *****************************
skipping: [104.236.244.212] => (item=[]) 

TASK [Add Ondrej Sury's apt key (Debian).] *************************************
skipping: [104.236.244.212]

TASK [Add Ondrej Sury's repo (Debian).] ****************************************
skipping: [104.236.244.212]

TASK [Update apt caches after repo is added (Debian).] *************************
skipping: [104.236.244.212]

TASK [Purge PHP version packages.] *********************************************
fatal: [104.236.244.212]: FAILED! => {"failed": true, "msg": "'php_packages' is undefined"}
	to retry, use: --limit @/home/ubuntu/dashboard/vendor/geerlingguy/drupal-vm/provisioning/playbook.retry

PLAY RECAP *********************************************************************
104.236.244.212            : ok=12   changed=0    unreachable=0    failed=1   


DRUPALVM_ENV=prod ansible-playbook -i vm/inventory vendor/geerlingguy/drupal-vm/provisioning/playbook.yml -e "config_dir=$(pwd)/vm" --sudo --tags=drupal --extra-vars "ansible_become_pass=${ANSIBLE_VAULT_PASSWORD}" returned exit code 2

Action failed: DRUPALVM_ENV=prod ansible-playbook -i vm/inventory vendor/geerlingguy/drupal-vm/provisioning/playbook.yml -e "config_dir=$(pwd)/vm" --sudo --tags=drupal --extra-vars "ansible_become_pass=${ANSIBLE_VAULT_PASSWORD}"
```

### Summary

Updated to latest version. The purging of PHP packages fails due to `"'php_packages' is undefined"`. I tried to manually define copying what the whole provides:

```
+php_packages: "{{ __php_packages | list }}"
```

Which returns, then:

```
TASK [Purge PHP version packages.] ***********************************************************************************************************************************************************************************************************
fatal: [104.236.244.212]: FAILED! => {"failed": true, "msg": "{{ __php_packages | list }}: '__php_packages' is undefined"}
	to retry, use: --limit @/Users/mglaman/Clients/commerceguys/reports-leancom/vendor/geerlingguy/drupal-vm/provisioning/playbook.retry
```

**Comment by mglaman** [comment]
This is failing locally and in CircleCI for same error.

**Comment by oxyc** [comment]
Seems you dont have https://github.com/geerlingguy/drupal-vm/commit/a5a9238080a09ab333cb9fc8aff23656738d72d3

**Comment by mglaman** [comment]
Here's my playbook. That seems to be there :/

```
---
- hosts: all
  become: yes

  vars_files:
    - vars/main.yml
    - ../default.config.yml

  pre_tasks:
    - name: Include OS-specific variables.
      include_vars: "{{ ansible_os_family }}.yml"

    - name: Define config_dir.
      set_fact:
        config_dir: "{{ playbook_dir }}/.."
      when: config_dir is not defined
      tags: ['always']

    - name: Include optional configuration files.
      include_vars: "{{ item }}"
      with_fileglob:
        - "{{ config_dir }}/config.yml"
        - "{{ config_dir }}/local.config.yml"
        - "{{ config_dir }}/secrets.yml"
        - "{{ config_dir }}/{{ lookup('env', 'DRUPALVM_ENV')|default(drupalvm_env, true)|default(ansible_env.DRUPALVM_ENV)|default(omit) }}.config.yml"
      tags: ['always']

    - include: tasks/hostname.yml
      tags: ['php', 'drupal']

    - include: tasks/init-debian.yml
      when: ansible_os_family == 'Debian'
      tags: ['php', 'drupal']
      static: no
    - include: tasks/init-redhat.yml
      when: ansible_os_family == 'RedHat'
      tags: ['php', 'drupal']
      static: no

    - name: Run configured pre-provision shell scripts.
      script: "{{ item }}"
      with_items: "{{ pre_provision_scripts|default([]) }}"

    - name: Run configured pre-provision ansible task files.
      include: "{{ outer_item }}"
      loop_control:
        loop_var: outer_item
      with_fileglob:
        - "{{ pre_provision_tasks_dir|default(omit) }}"

    - include: tasks/php.yml
      tags: ['php', 'drupal']

  roles:
    # Essential roles.
    - { role: geerlingguy.repo-remi, when: ansible_os_family == 'RedHat', tags: ['webserver, php'] }
    - { role: geerlingguy.firewall }
    - { role: geerlingguy.git }
    - { role: geerlingguy.postfix }
    - { role: geerlingguy.apache, when: drupalvm_webserver == 'apache', tags: ['webserver']}
    - { role: geerlingguy.apache-php-fpm, when: drupalvm_webserver == 'apache', tags: ['webserver'] }
    - { role: geerlingguy.nginx, when: drupalvm_webserver == 'nginx', tags: ['webserver'] }
    - { role: geerlingguy.php, tags: ['php', 'drupal'] }
    - { role: geerlingguy.php-pecl, tags: ['php'] }
    - { role: geerlingguy.composer }
    - { role: geerlingguy.mysql, when: drupal_db_backend == 'mysql', tags: ['database'] }
    - { role: geerlingguy.php-mysql, when: drupal_db_backend == 'mysql', tags: ['php', 'database'] }
    - { role: geerlingguy.postgresql, when: drupal_db_backend == 'pgsql', tags: ['database'] }
    - { role: geerlingguy.php-pgsql, when: drupal_db_backend == 'pgsql', tags: ['php', 'database'] }

    # Conditionally-installed roles.
    - { role: geerlingguy.drupal-console, when: 'drupal_major_version > 7 and "drupalconsole" in installed_extras' }
    - { role: geerlingguy.drush, when: '"drush" in installed_extras' }
    - { role: geerlingguy.memcached, when: '"memcached" in installed_extras' }
    - { role: geerlingguy.php-memcached, when: '"memcached" in installed_extras' }

    - role: geerlingguy.php-tideways
      workspace: "/root/php{{ php_version }}"
      when: '"tideways" in installed_extras'

    - role: geerlingguy.php-xdebug
      workspace: "/root/php{{ php_version }}"
      when: '"xdebug" in installed_extras'
      tags: ['xdebug']

    - role: geerlingguy.php-xhprof
      workspace: "/root/php{{ php_version }}"
      when: '"xhprof" in installed_extras'

    - role: thom8.php-upload-progress
      workspace: "/root/php{{ php_version }}"
      when: '"upload-progress" in installed_extras'

    - { role: geerlingguy.blackfire, when: '"blackfire" in installed_extras' }
    - { role: geerlingguy.adminer, when: '"adminer" in installed_extras', tags: ['database'] }
    - { role: geerlingguy.pimpmylog, when: '"pimpmylog" in installed_extras' }
    - { role: geerlingguy.daemonize, when: '"mailhog" in installed_extras' }
    - { role: geerlingguy.mailhog, when: '"mailhog" in installed_extras' }
    - { role: franklinkim.newrelic, when: '"newrelic" in installed_extras' }
    - { role: geerlingguy.nodejs, when: '"nodejs" in installed_extras' }
    - { role: geerlingguy.redis, when: '"redis" in installed_extras' }
    - { role: geerlingguy.php-redis, when: '"redis" in installed_extras' }
    - { role: geerlingguy.ruby, when: '"ruby" in installed_extras' }

    - role: geerlingguy.java
      when: >
        "java" in installed_extras or
        "solr" in installed_extras or
        "selenium" in installed_extras or
        "elasticsearch" in installed_extras

    - { role: arknoll.selenium, when: '"selenium" in installed_extras' }
    - { role: geerlingguy.solr, when: '"solr" in installed_extras' }
    - { role: geerlingguy.elasticsearch, when: '"elasticsearch" in installed_extras' }
    - { role: geerlingguy.varnish, when: '"varnish" in installed_extras' }
    - { role: drupalvm.www, tags: ['webserver'] }
    - { role: geerlingguy.drupal, tags: ['drupal'] }

    # Roles for security and stability on production.
    - { role: geerlingguy.security, when: extra_security_enabled }

  tasks:
    - include: tasks/sshd.yml
    - include: tasks/extras.yml

    - include: tasks/apparmor.yml
      when: ansible_os_family == 'Debian' and drupal_db_backend == 'mysql'
      tags: ['database']

    - include: tasks/drush-aliases.yml

    - include: tasks/cron.yml
      tags: ['cron']

    - include: tasks/dashboard.yml
      when: dashboard_install_dir is defined and dashboard_install_dir != ''
      tags: ['webserver', 'database']

    - name: Run configured post-provision shell scripts.
      script: "{{ item }}"
      with_items: "{{ post_provision_scripts|default([]) }}"

    - name: Run configured post-provision ansible task files.
      include: "{{ outer_item }}"
      loop_control:
        loop_var: outer_item
      with_fileglob:
        - "{{ post_provision_tasks_dir|default(omit) }}"

```

**Comment by oxyc** [comment]
Your log doesn't mention it though hmm...

```
TASK [Gathering Facts] *********************************************************
ok: [104.236.244.212]

TASK [Define config_dir.] ******************************************************
skipping: [104.236.244.212]
```

**Comment by mglaman** [comment]
Weird! That's the CircleCI log. Local has this.

```
TASK [Include OS-specific variables.] ********************************************************************************************************************************************************************************************************
ok: [104.236.244.212]
```

Now that I know the task I'll see what I can crop up.

**Comment by mglaman** [comment]
Yep. It's not running that `Include OS-specific variables.` task. Almost like it's using an old playbook? but it is using new one.

**Comment by mglaman** [comment]
It doesn't run when using tags, PR #1320.

**Comment by oxyc** [comment]
doh... my bad

**Comment by geerlingguy** [comment]
Fixed by #1320!
